packages/workbox-cli/src/lib/questions/ask-extensions-to-cache.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+const assert = require('assert');
+const inquirer = require('inquirer');
+const nodeDir = require('node-dir');
+const ora = require('ora');
+const path = require('path');
+
+const errors = require('../errors');
+const {ignoredDirectories} = require('../constants');
+
+// The key used for the question/answer.
+const name = 'globPatterns';
+
+/**
+ * @param {string} globDirectory The directory used for the root of globbing.
+ * @return {Promise<[string]>} The unique file extensions corresponding to all
+ * of the files under globDirectory.
+ */
+async function getAllFileExtensions(globDirectory) {
+  const substringsToIgnore = ignoredDirectories.map(
+    (directory) => `${path.sep}${directory}${path.sep}`);
+
+  const allFiles = await nodeDir.promiseFiles(globDirectory, 'file',
+    {shortName: false, recursive: true});
+
+  const filteredFiles = allFiles.filter(
+    (filePath) => substringsToIgnore.every(
+      (substring) => !filePath.includes(substring))
+  );
+
+  const extensions = new Set();
+  for (const file of filteredFiles) {
+    const extension = path.extname(file);
+    if (extension) {
+      // Get rid of the leading . character.
+      extensions.add(extension.replace(/^\./, ''));
+    }
+  }
+
+  return [...extensions];
+}
+
+/**
+ * @param {string} globDirectory The directory used for the root of globbing.
+ * @return {Promise<Object>} The answers from inquirer.
+ */
+async function askQuestion(globDirectory) {
+  const spinner = ora({
+    text: `Examining files in ${globDirectory}...`,
+    stream: process.stdout,
+  }).start();
+  const fileExtensions = await getAllFileExtensions(globDirectory);
+  spinner.stop();
+
+  assert(fileExtensions.length > 0, errors['no-file-extensions-found']);
```

Comment:
Just tested assert on node and the output from this isn't the most friendly thing:

> assert(false, 'test')
AssertionError [ERR_ASSERTION]: test
    at repl:1:1
    at ContextifyScript.Script.runInThisContext (vm.js:50:33)
    at REPLServer.defaultEval (repl.js:240:29)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:441:10)
    at emitOne (events.js:120:20)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:282:10)
    at REPLServer.Interface._line (readline.js:631:8)

Reply:
We [clean up any exceptions](https://github.com/GoogleChrome/workbox/blob/v3/packages/workbox-cli/src/lib/cleanup-stack-trace.js) prior to showing them to the user. The failure here looks like:

```sh
$ ./bin.js wizard
? What is the root of your web app? Manually enter path
? Path to your web app's root: /tmp/a

No files could be found that are suitable for caching.
```

(with the last line in red).

Reply:
So you are removing the `AssertionError [ERR_ASSERTION]:` in the log as well somehow?

Reply:
Yes, when there's an exception, the [CLI logs the `message` property along with the cleaned-up stacktrace](https://github.com/GoogleChrome/workbox/blob/v3/packages/workbox-cli/src/bin.js#L38-L39). The `message` property of an assertion failure just uses the string that was passed in:

```sh
> try { assert(false, 'this is a test'); } catch(e) { console.error(e.message); }
this is a test
```

When the CLI is used with `--debug`, we log the raw stracktrace, which would lead to the `AssertionError [ERR_ASSERTION]:` text, but that's not what users would see by default.